### PR TITLE
gokrb5: add support for Go modules

### PR DESCRIPTION
**Pull request by sbinet**
this CL is equivalent to #185 but for the new `v6` branch.

### CODE REVIEW

**Comment by jcmturner** [comment]
Thanks for this. I'm still reading up on Go modules and will review once I have done that.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module gopkg.in/jcmturner/gokrb5.v6
```
[the remark this answers is not in the document]
Comment (sbinet) [reply go.mod]: For the v6 branch, it should be actually:

```
module github.com/jcmturner/gokrb5/v6
```

**Comment by jcmturner** [comment]
Hi @sbinet I've finally got round to sorting out Go modules. I have just released v8 which is implemented as a module. Sorry for the delay but this supersedes this PR so I'm going to close it. Thanks for the contribution and I hope v8 is what you need.

**Comment by sbinet** [comment]
I am a bit surprised by your `/v8` subdir and with keeping the old code around.

`gopkg.in` should have handled the legacy code just fine (tugged away at the `v7.x.y` branch) and one could just put the module at the root of the repository.

but perhaps I am missing something?

**Comment by jcmturner** [comment]
Reading this: https://blog.golang.org/v2-go-modules
It advises that major versions >=v2 should be in their own sub-directory.

**Comment by sbinet** [comment]
yes, but as the explicit reason that is being described in that document explaining why this could be done in that way (_ie:_ adding a sub directory `/vX` to not break packages compiled with Go versions non-module-aware (v<1.13)) is already handled by the `gopkg.in/...vXX` import path, this repository (and its "new", Go-module aware import path `"github.com/jcmturner/gokrb5/vXX"`) could do w/o that "hack".

anyways. that's a minor cosmetic point.

thanks for that package! :)

**Pull request closed without merging**